Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -942,23 +967,33 @@ protected override void ProcessRecord()
                 // Get the underlying byte representation of the string using the same encoding (UTF-16).
                 // So the data will be written in the same format it is passed, rather than converting to
                 // UTF-8 or UTF-32 when writen to Cloud Storage.
-                byte[] contentBuffer = Encoding.Unicode.GetBytes(Contents ?? "");
+                byte[] contentBuffer = Encoding.Unicode.GetBytes(Value ?? "");
                 contentStream = new MemoryStream(contentBuffer);
             }
 
             // Get the existing storage object so we can use its metadata. (If it does not exist, we will fall back to
             // default values.)
-            Object existingGcsObject = null;
+            Object existingGcsObject = InputObject;
             Dictionary<string, string> existingObjectMetadata = null;
 
             using (contentStream)
             {
                 try
                 {
-                    ObjectsResource.GetRequest getReq = Service.Objects.Get(Bucket, ObjectName);
-                    getReq.Projection = ObjectsResource.GetRequest.ProjectionEnum.Full;
+                    if (existingGcsObject == null)
+                    {
+                        ObjectsResource.GetRequest getReq = Service.Objects.Get(Bucket, ObjectName);
+                        getReq.Projection = ObjectsResource.GetRequest.ProjectionEnum.Full;
+
+                        existingGcsObject = getReq.Execute();
+                    }
+                    else
+                    {
+                        // Set these variables so the call to UploadGcsObject at the end of the function will succeed.
+                        Bucket = existingGcsObject.Bucket;
+                        ObjectName = existingGcsObject.Name;
```

Comment:
Instead of doing this, you should use existingGcsObject as a single source of data at the UploadGcsObject call.